test/Microsoft.AspNetCore.Mvc.FunctionalTests/JsonPatchInputFormatterTest.cs

```diff
@@ -0,0 +1,136 @@
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var body = await response.Content.ReadAsStringAsync();
+            var product = JsonConvert.DeserializeObject<Product>(body);
+            Assert.NotNull(product);
+            Assert.NotNull(product.Reviews);
+            Assert.Equal(2, product.Reviews.Count);
+            Assert.Equal(5, product.Reviews[0].Rating);
+        }
+
+        [Fact]
+        public async Task CopyOperation_Works()
+        {
+            // Arrange
+            var input = "[{ 'op': 'copy', 'path': 'Reviews/1/Rating', 'from': 'Reviews/0/Rating'}]".Replace("'", "\"");
+            var request = GetPatchRequest(input);
+
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var body = await response.Content.ReadAsStringAsync();
+            var product = JsonConvert.DeserializeObject<Product>(body);
+            Assert.NotNull(product);
+            Assert.NotNull(product.Reviews);
+            Assert.Equal(2, product.Reviews.Count);
+            Assert.Equal(4, product.Reviews[1].Rating);
+            Assert.Equal(4, product.Reviews[0].Rating);
+        }
+
+        [Fact]
+        public async Task MoveOperation_Works()
+        {
+            // Arrange
+            var input = "[{ 'op': 'move', 'path': 'Reviews/1/Rating', 'from': 'Reviews/0/Rating'}]".Replace("'", "\"");
+            var request = GetPatchRequest(input);
+
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var body = await response.Content.ReadAsStringAsync();
+            var product = JsonConvert.DeserializeObject<Product>(body);
+            Assert.NotNull(product);
+            Assert.NotNull(product.Reviews);
+            Assert.Equal(2, product.Reviews.Count);
+            Assert.Equal(4, product.Reviews[1].Rating);
+            Assert.Equal(0, product.Reviews[0].Rating);
+        }
+
+        [Fact]
+        public async Task RemoveOperation_Works()
+        {
+            // Arrange
+            var input = "[{ 'op': 'remove', 'path': 'Reviews/0/Rating'}]".Replace("'", "\"");
+            var request = GetPatchRequest(input);
+
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var body = await response.Content.ReadAsStringAsync();
+            var product = JsonConvert.DeserializeObject<Product>(body);
+            Assert.NotNull(product);
+            Assert.NotNull(product.Reviews);
+            Assert.Equal(2, product.Reviews.Count);
+            Assert.Equal(0, product.Reviews[0].Rating);
+        }
+
+        private HttpRequestMessage GetPatchRequest(string body)
```

Comment:
We should include one case where there are some errors as a result of the operation. This has way more integration risk than the successful operations.